adaptdl/adaptdl/torch/adascale.py

```diff
@@ -268,8 +280,8 @@ def _final_callback(self):
         if count > 1:
             grad_sqr = (count * total_sqr - local_sqr) / (count - 1)
             grad_var = (local_sqr - total_sqr) * scale / (count - 1)
-            self.raw_grad_sqr = grad_sqr
-            self.raw_grad_var = grad_var
+            grad_sqr = np.maximum(grad_sqr, 0.0)
+            grad_var = np.maximum(grad_var, 1e-6)
```

Comment:
We found that clipping post-smoothing performs better than clipping pre-smoothing, so should probably remove these two lines.